File: 2582-pass-the-pillow/main.rs
impl Solution {
    pub fn pass_the_pillow(n: i32, time: i32) -> i32 {
        
        let (mut position, mut direction) = (1,  1);

        for _ in 0..time{
            position += direction;
            if position == n {
                direction = -1;
            }else if position == 1 {
                direction = 1;
            }
        }

        position
    }
}

File: 2582-pass-the-pillow/main.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
        
        int cycle_length = 2 * (n - 1);
        int effective_time = time % cycle_length;

        if (effective_time < n) {
            return 1 + effective_time;
        }

        return 2 * n - 1 - effective_time;
        
    }
};
